crates/codegen/spec/src/public/mod.rs

```diff
@@ -0,0 +1,126 @@
+mod markdown;
+mod navigation;
+
+use std::path::PathBuf;
+
+use codegen_schema::types::grammar::{Grammar, GrammarSection, GrammarTopic};
+use codegen_utils::context::CodegenContext;
+
+use crate::{
+    public::{markdown::MarkdownWriter, navigation::NavigationEntry},
+    snippets::Snippets,
+};
+
+pub struct PublicPages<'a> {
+    grammar: &'a Grammar,
+    snippets: &'a Snippets<'a>,
+}
+
+impl<'a> PublicPages<'a> {
+    pub fn new(grammar: &'a Grammar, snippets: &'a Snippets<'a>) -> Self {
+        return Self { grammar, snippets };
+    }
+
+    pub fn write_to_disk(
```

Comment:
Not so keen on this name ... `disk` ? Maybe not a disk ...

Reply:
Changed to `write_files`. How about that?